classes/XDUser.php

```diff
@@ -3044,4 +3158,174 @@ public static function validateRID($rid)
 
         return $results;
     }
+
+    /**
+     * Executes any actions that need to be conducted immediately after a user is logged into XDMoD.
+     *
+     * @throws Exception if there is a problem executing any of the required post logged in steps.
+     */
+    public function postLogin() {
+        if (!$this->isSticky()) {
+            $this->updatePerson();
+            $this->synchronizeOrganization();
+        }
+        $this->currentToken = XDSessionManager::recordLogin($this);
+    }
+
+    /**
+     * Attempts to synchronize this users organization. This will ensure that if the organization
+     * assigned to the user is not the same as the organization that should be assigned to the user
+     * ( i.e. the organization assigned to this users person has been updated ). They have their
+     * organization updated. If they have been associated with a 'center' related acl, center
+     * director or center staff, then they are to have these acls removed and an email notice sent
+     * to the admin / user notifying them that additional steps will need to be taken before their
+     * former level of access is restored.
+     *
+     * @throws Exception
+     */
+    public function synchronizeOrganization()
+    {
+        // This is pulled from the moddb.Users.organization_id column for this user record.
+        $actualOrganization = $this->getOrganizationID();
+
+        // Retrieve the organization associated with this users Person. This is the value we expect
+        // the user's organization to match.
+        $expectedOrganization = Organizations::getOrganizationIdForPerson($this->getPersonID());
+
+        // If we have ssoAttrs available and this user's person's organization is 'Unknown' ( -1 ).
+        // Then go ahead and lookup the organization value from sso.
+        if (count($this->ssoAttrs) > 0 && $expectedOrganization === -1) {
```

Comment:
Micro optimization to allow the conditional to not bother to count if the organization isnt -1
```suggestion
        if ($expectedOrganization === -1 && count($this->ssoAttrs) > 0) {
```